### ISSUE TITLE
ARM64 version?

### ISSUE THREAD

**Issue by Lightwel** [gehelem/als#161]
Hi. I have ALS running on my laptop and it looks simple and elegant. 

I would like to run it on RaspberryPi 64 OS on a Pi 4 8Gb with CCDCiel for capture.

But both the Linux and the RPi executables from nightly builds do not run with ./ .

I am guessing the Linux is X86/64 and the pi is ARMhf.

Can we have an ARM64 version pls? 

For me I would prefer the simple Linux version not the whole py env package, but others may differ.

If my guess is wrong and the executable should run from $ ./, any hints? 

Tks R

**Comment by gehelem** [comment]
Hi. thank you for the report
Can you provide more details ?
(command line output ...)

(i let @deufrai  answering your ARM64 question)

G.

**Comment by Lightwel** [comment]
Hello and thanks for responding.

This is the cmd line output (bold added for emphasis):

***@***.***:~ $ cd Downloads
***@***.***:~/Downloads $ ls
als                      build-essential_12.9_arm64.deb  h17_star_database_mag17_astap.deb  libpasastro-1.4.1-51-linux_armhf.tar.xz
als-0.7-dev-7533ab0.run  ccdciel_0.9.77-2611_arm64.deb   h18_star_database_mag18_astap.deb  Nou
astap_arm64.deb          eqmodgui_1.7.0-127_arm64.deb    hnsky_arm64.deb                    skychart_4.3-4466_arm64.deb
astap_arm64_qt5.tar.gz   FireCapture_v2.6_armhf.tar.gz   hnsky_home_documents.zip
astap_arm64.tar.gz       g17_star_database_mag17.zip     indistarter_2.3.0-159_arm64.deb
***@***.***:~/Downloads $ ./als-0.7-dev-7533ab0.run
bash: ./als-0.7-dev-7533ab0.run: cannot execute binary file: Exec format error
***@***.***:~/Downloads $ ls -l als-0.7-dev-7533ab0.run
-rwxr-xr-x 1 pi pi 154244152 Mar 14 06:18 als-0.7-dev-7533ab0.run
***@***.***:~/Downloads $ 

Tks

R

**Comment by gehelem** [comment]
Okay
Image file for RPi is compressed  (*.tgz)
you should unzip its content somewhere before.
Then you can launch ALS with this :
![image](https://user-images.githubusercontent.com/20287008/158580859-2b4a1dd3-c2f2-4c9c-95fe-72506c546ff3.png)
Please note this is specific to RPi
Tell us if it works
G.

**Comment by Lightwel** [comment]
Tks: it doesn’t

**Comment by Lightwel** [comment]
This pic is the message from Dolphin when seeking to run the exec in the .tar.

Best 
R




Sent from my iPhone

**Comment by gehelem** [comment]
sorry i don't understand your answer ...
you must extract the content of the tgz file somewhere and _then_ launch the executable file.
(don't run it directly into Ark or any compression software)
is this what you do ?

**Comment by deufrai** [comment]
hi

`./als-0.7-dev-7533ab0.run: cannot execute binary file: Exec format error`

trying to execute a .run file on a Pi will fail
Those file are intended for AMD/Intel Linux systems

if you want to run ALS on a Pi, you have to be in 32 bits or wait a bit more :) Sorry

**Comment by Lightwel** [comment]
Ok, understand re the .run. That’s what I suspected.

Executing the file extracted from the tar gives a similar error as per the screenshot in my 2nd email earlier (sorry failed to attach it first time). 

Looking forward to the 64bit version.

Tks
R

**Comment by deufrai** [comment]
Hi again

To clarify what seems to be a misunderstanding:
All recent and stable enough builds can be found here : https://als-app.org/nightlies/

.run filed are for Intel or AMD 64 bit Linux
.tgz files are for arm 32 bit Linux

I'll change that web page soon enough to make this more obvious

As soon as I can get my hands on a new raspberry pi, I'll work on being also able to build and publish arm 64 bit Linux versions

I'll comment on this ticket as soon as it is ready

In the meantime, since you can use ALS on your laptop, your ideas and feedback are most welcome ! 😊

Clear skies !

**Comment by Lightwel** [comment]
Sorry - that should have been obvious from what you said previously.

Thanks for the work on the new release as and when.

Best

R

**Comment by deufrai** [comment]
Really, nothing for you to be sorry for.
OUR webpages are lacking important info.

Your ticket helped us realize that.

**Comment by deufrai** [comment]
Hi

A ready-to-use archive for 64bits Raspberry Pis is finally ready for consumption.
Please refer to : https://github.com/gehelem/als/issues/189#issuecomment-2176544779

Please let us know if it runs OK on your system

**Comment by deufrai** [comment]
Closing this issue as completed. 
ARM64 archive is now published on our latest release page : https://github.com/gehelem/als/releases/tag/v0.7-beta2
